cmd/ketch/app.go

```diff
@@ -26,3 +31,80 @@ func newAppCmd(cfg config, out io.Writer) *cobra.Command {
 	cmd.AddCommand(newAppExportCmd(cfg, out, appExport))
 	return cmd
 }
+
+func appState(pods []apiv1.Pod) (state string) {
+	state = "unknown"
+	// If there's no pods for this app, then app in `created` state
+	if len(pods) == 0 {
+		state = strings.ToLower(string(ketchv1.AppCreated))
+		return state
+	}
+
+	// If app has pods, then we build app status based on pods statuses
+	var deploying, running, errored, succeeded int
+	for _, pod := range pods {
+		switch podState(pod) {
+		case ketchv1.PodRunning:
+			running++
+		case ketchv1.PodDeploying:
+			deploying++
+		case ketchv1.PodError:
+			errored++
+		case ketchv1.PodSucceeded:
+			succeeded++
+		}
+	}
+
+	var parts []string
+	if deploying > 0 {
+		parts = append(parts, fmt.Sprintf(`%d %s`, deploying, ketchv1.PodDeploying))
+	}
+	if running > 0 {
+		parts = append(parts, fmt.Sprintf(`%d %s`, running, ketchv1.PodRunning))
+	}
+	if errored > 0 {
+		parts = append(parts, fmt.Sprintf(`%d %s`, errored, ketchv1.PodError))
+	}
+	if succeeded > 0 {
+		parts = append(parts, fmt.Sprintf(`%d %s`, succeeded, ketchv1.PodSucceeded))
+	}
+
+	return strings.Join(parts, ", ")
+}
+
+var (
+	containerStateCrashLoopBackOff = "CrashLoopBackOff"
+	containerStateCompleted        = "Completed"
+)
+
+func podState(pod apiv1.Pod) ketchv1.PodState {
```

Comment:
I think we should discuss this idea of having `ketchv1.PodState`, it is a refined version of `apiv1.PodPhase`. 
A summary of the refinements is:
1) rename PodPending to PodDeploying
2) rename PodFailed to PodError 
3) override a pod's PodRunning if something has happened with one of the pod's container. 

Do we need 1 and 2? 
People got used to kubernetes Pod State.

Reply:
Well, had the same question in my mind, this is the product requirements, not mine

Reply:
at least we can implement something like

```golang

func refinePodState(pod apiv1.Pod) apiv1.PodPhase {
	switch pod.Status.Phase {
	case apiv1.PodPending:
		return apiv1.PodPending
	case apiv1.PodRunning:
		// In some cases, we have to override the status obtained by using `pod.Status.Phase` because it could be
		// `running` state while the actual container in `CrashLoopBackOff` so, actual state of the pod is `error`,
		// and so on.
		for _, c := range pod.Status.ContainerStatuses {
			// If any of the pod containers in the `CrashLoopBackOff` state we must return the `error` state
			if c.State.Waiting != nil {
				if c.State.Waiting.Reason == containerStateCrashLoopBackOff {
					return apiv1.PodFailed
				}
			}
			// If any of the pod containers in the `Completed` state we must return the `succeeded` state
			if c.State.Terminated != nil {
				if c.State.Terminated.Reason == containerStateCompleted {
					return apiv1.PodSucceeded
				}
			}
		}
		return apiv1.PodRunning
	case apiv1.PodFailed, apiv1.PodUnknown:
		return apiv1.PodFailed
	case apiv1.PodSucceeded:
		return apiv1.PodSucceeded
	default:
		return apiv1.PodFailed
	}
}

```
but I suggest discussing the task's requirements

Reply:
Well, in this case, we will use "apiv1" statuses but will mutate the original state. So the state in Ketch will be different from the state in Kubernetes while naming be the same. So, this will confuse the user (it will confuse even me with time), because you need to know that the state in ketch is different from the state in Kubernetes. Sure, we may assume that users know all ketch documentation, but they will not, and we don't have such good documentation, yet. 

Separate naming is better in this case, IMHO.